src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -958,25 +958,55 @@ public static Vec3d rayTraceEyeHitVec(EntityLivingBase entity, double length)
         return git == null ? null : git.hitVec;
     }
 
+    @Deprecated
     public static boolean onInteractEntityAt(EntityPlayer player, Entity entity, RayTraceResult ray, ItemStack stack, EnumHand hand)
     {
-        Vec3d vec3d = new Vec3d(ray.hitVec.xCoord - entity.posX, ray.hitVec.yCoord - entity.posY, ray.hitVec.zCoord - entity.posZ);
-        return onInteractEntityAt(player, entity, vec3d, stack, hand);
+        return onInteractEntityAtAction(player, entity, ray, stack, hand) != null;
     }
 
+    @Deprecated
     public static boolean onInteractEntityAt(EntityPlayer player, Entity entity, Vec3d vec3d, ItemStack stack, EnumHand hand)
     {
-        return MinecraftForge.EVENT_BUS.post(new PlayerInteractEvent.EntityInteractSpecific(player, hand, stack, entity, vec3d));
+        return onInteractEntityAtAction(player, entity, vec3d, stack, hand) != null;
+    }
+
+    public static EnumActionResult onInteractEntityAtAction(EntityPlayer player, Entity entity, RayTraceResult ray, ItemStack stack, EnumHand hand)
+    {
+        Vec3d vec3d = new Vec3d(ray.hitVec.xCoord - entity.posX, ray.hitVec.yCoord - entity.posY, ray.hitVec.zCoord - entity.posZ);
+        return onInteractEntityAtAction(player, entity, vec3d, stack, hand);
+    }
+
+    public static EnumActionResult onInteractEntityAtAction(EntityPlayer player, Entity entity, Vec3d vec3d, ItemStack stack, EnumHand hand)
+    {
+        PlayerInteractEvent.EntityInteractSpecific evt = new PlayerInteractEvent.EntityInteractSpecific(player, hand, stack, entity, vec3d);
+        MinecraftForge.EVENT_BUS.post(evt);
+        return evt.isCanceled() ? evt.getSubstituteResult() : null;
```

Comment:
If you had looked at where these methods are called, you'd know that `null` is a perfectly valid value to return, it means to revert to vanilla behavior.